### ISSUE TITLE
Grab user's Email during authentication with Twitter

### ISSUE THREAD

**Issue by halfstrik** [omab/python-social-auth#825]
Greetings!

I perform a little googling and find out that Twitter recently adds optional returning of users Email address.
Look at the end of this long [thread](https://twittercommunity.com/t/how-to-get-email-from-twitter-user-using-oauthtokens/558/193) and [Twitter docs](https://dev.twitter.com/rest/reference/get/account/verify_credentials).

In short, if your app is approved by Twitter and you provide links to ToS and Privacy Policy for your site, you can (as they claim) get a user email address by providing `include_email` in request to endpoint `https://api.twitter.com/1.1/account/verify_credentials.json`.

Does any one succeed at this? Any thoughts?

**Comment by hellvix** [comment]
I submitted a request to Twitter asking for more user details (including e-mail) for my app. They approved, I changed social-auth's parameters and I am still unable to get the e-mail in the request. 
It would be interesting to know if someone succeeded in this.

**Comment by halfstrik** [comment]
I managed to get it right and actually receive user email!

For this several steps need to be done ([see this post](https://twittercommunity.com/t/how-to-get-email-from-twitter-user-using-oauthtokens/558/209?u=emergeworld))

Make sure Twitter respond to you via email and approved that your app actually get white-listed.

Than this changes need to be done in python-social-auth app in `python/lib/python2.7/site-packages/social/backends/twitter.py`:
line 30: `'email': response.get('email'),`
line 38: `'https://api.twitter.com/1.1/account/verify_credentials.json?include_email=true',`

In other words, `Parameters` from Twitter documentation must go not to `access_token` object, but rather to URL as path params.

**Comment by hellvix** [comment]
It worked, @halfstrik. It should be fixed in the app.

**Comment by halfstrik** [comment]
@hellvix, great to hear it! I updated appropriate docs/ files in this PR as well.

The things I'm unsure are the name of the variable in settings and am I write a proper test.
Will wait @omab for judgments:)

**Comment by omab** [comment]
Fix already merged, thanks!
